fastlane/lanes/release.rb

```diff
@@ -104,30 +107,44 @@
 
     push_to_git_remote(tags: false)
 
-    attempts = 0
+    # Protect release/* branch
+    copy_branch_protection(
+      repository: GITHUB_REPO,
+      from_branch: DEFAULT_BRANCH,
+      to_branch: release_branch_name
+    )
+
     begin
-      attempts += 1
-      copy_branch_protection(
+      # Move PRs to next milestone
+      moved_prs = update_assigned_milestone(
+        repository: GITHUB_REPO,
+        from_milestone: new_version,
+        to_milestone: release_version_next,
+        comment: "Version `#{new_version}` has now entered code-freeze, so the milestone of this PR has been updated to `#{release_version_next}`."
+      )
+
+      # Add ❄️ marker to milestone title to indicate we entered code-freeze
+      set_milestone_frozen_marker(
         repository: GITHUB_REPO,
-        from_branch: DEFAULT_BRANCH,
-        to_branch: release_branch_name,
-        github_token: get_required_env('GITHUB_TOKEN')
+        milestone: new_version
       )
     rescue StandardError => e
-      if attempts < 2
-        sleep_time = 5
-        UI.message("Failed to set branch protection on GitHub. Retrying in #{sleep_time} seconds in case it was because the API hadn't noticed the new branch yet.")
-        sleep(sleep_time)
-        retry
-      else
-        UI.error("Failed to set branch protection on GitHub after #{attempts} attempts")
-        raise e
-      end
+      moved_prs = []
+
+      report_milestone_error(error_title: "Error freezing milestone `#{new_version}`: #{e.message}")
     end
 
-    set_milestone_frozen_marker(repository: GITHUB_REPO, milestone: new_version)
+    UI.message("Moved the following PRs to milestone #{release_version_next}: #{moved_prs.join(', ')}")
 
-    check_pods_references
+    # Annotate the build with the moved PRs
+    moved_prs_info = if moved_prs.empty?
+                       "👍 No open PR were targeting `#{new_version}` at the time of code-freeze"
```

Comment:
Nitpick: were requires a plural

```suggestion
                       "👍 No open PRs were targeting `#{new_version}` at the time of code-freeze"
```